src/module.c

```diff
@@ -36,7 +36,7 @@ JL_DLLEXPORT jl_module_t *jl_new_module(jl_sym_t *name)
     htable_new(&m->bindings, 0);
     arraylist_new(&m->usings, 0);
     JL_GC_PUSH1(&m);
-    if (jl_core_module) {
+    if (jl_core_module && using_core) {
         jl_module_using(m, jl_core_module);
     }
     // export own name, so "using Foo" makes "Foo" itself visible
```

Comment:
We might want to omit this as well, just so you can get a truly truly empty module.

Reply:
Triage seconds this.

Reply:
And then I guess `using_core` should be renamed `default_names` or something like that? Maybe not important.